test/Microsoft.AspNet.Razor.Runtime.Test/DefaultTagHelperContentTest.cs

```diff
@@ -0,0 +1,388 @@
+
+        [Fact]
+        public void CanIdentifyIsEmpty_Append_TagHelperContent()
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+            var copiedTagHelperContent = new DefaultTagHelperContent();
+
+            // Act
+            tagHelperContent.Append(copiedTagHelperContent);
+            tagHelperContent.Append(string.Empty);
+
+            // Assert
+            Assert.True(tagHelperContent.IsEmpty);
+        }
+
+        // Clear
+        [Fact]
+        public void CanClearContent()
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+            tagHelperContent.SetContent("Hello World");
+
+            // Act
+            tagHelperContent.Clear();
+
+            // Assert
+            Assert.True(tagHelperContent.IsEmpty);
+        }
+
+        // CopyTo
+        [Fact]
+        public void CanWriteToTextWriter()
+        {
+            // Arrange
+            var writer = new StringWriter();
+            var tagHelperContent = new DefaultTagHelperContent();
+            var expected = "Hello World!";
+            tagHelperContent.SetContent(expected);
+
+            // Act
+            tagHelperContent.CopyTo(writer);
+
+            // Assert
+            Assert.Equal(expected, writer.ToString());
+        }
+
+        [Fact]
+        public void CanWriteToTextWriter_MultipleAppends()
+        {
+            // Arrange
+            var writer = new StringWriter();
+            var tagHelperContent = new DefaultTagHelperContent();
+            var text1 = "Hello";
+            var text2 = " World!";
+            var expected = text1 + text2;
+            tagHelperContent.Append(text1);
+            tagHelperContent.Append(text2);
+
+            // Act
+            tagHelperContent.CopyTo(writer);
+
+            // Assert
+            Assert.Equal(expected, writer.ToString());
+        }
+
+        [Fact]
+        public void CanConvertToString()
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+            var expected = "Hello World!";
+            tagHelperContent.SetContent(expected);
+
+            // Act
+            var actual = tagHelperContent.ToString();
+
+            // Assert
+            Assert.Equal(expected, actual, StringComparer.Ordinal);
+        }
+
+        [Fact]
+        public void GetEnumerator_EnumeratesThroughBuffer()
+        {
+            // Arrange
+            var tagHelperContent = new DefaultTagHelperContent();
+            var expected = new string[] { "Hello", "World" };
+            tagHelperContent.SetContent(expected[0]);
+            tagHelperContent.Append(expected[1]);
+            var i = 0;
+
+            // Act & Assert
+            foreach (var val in tagHelperContent)
+            {
+                Assert.Equal(expected[i++], val);
+            }
+        }
+
+        [Fact]
+        public void SetContent_Fluent_WithAppend()
```

Comment:
- name the fluent tests consistently and include expectations e.g. `Fluent_SetContent_Append_WritesExpectedContent`
- leave out `Fluent_` unless it's required to make test name unique